llvm/lib/Target/AMDGPU/AsmParser/AMDGPUAsmParser.cpp

```diff
@@ -3039,7 +3038,8 @@ bool AMDGPUAsmParser::ParseAMDGPURegister(RegisterKind &RegKind, unsigned &Reg,
     if (Reg == AMDGPU::SGPR_NULL) {
       Error(Loc, "'null' operand is not supported on this GPU");
     } else {
```

Comment:
This `else` is probably not needed, `Error` shall act like a return.

Reply:
Actually it does not return, but can be used as `return Error()` instead.

Reply:
`MCAsmParser::Error()` seems to return true, so wouldn't work. (Confusingly, our `AMDGPUAsmParser::Error()` returns false, but from what I see that's not what we call here.)